src/grz_cli/upload.py

```diff
@@ -142,6 +142,7 @@ def upload_file(self, local_file_path, s3_object_id):
             multipart_threshold=MULTIPART_THRESHOLD,
             multipart_chunksize=chunksize,
             max_concurrency=self._threads,
+            use_threads=self._threads > 1,
```

Comment:
Ah was it not multithreading before when multiple threads were requested?

Reply:
Not sure, but this is at the very least more explicit.

Reply:
Checked, use_threads defaults to `True` (because max concurrency defaults to 10). So if threads == 1, this will indeed have an effect